solutions/APIO 11-path.mdx

```diff
@@ -2,13 +2,23 @@
 id: apio-11-path
 source: APIO 2011
 title: Find the Path
-author: Andi Qu
+author: Albert Ye, Andi Qu
 ---
 
-<IncompleteSection />
+# Solution
+The hardest part of this problem is designing the nodes and the edges to use in the Dijkstra. The framing of the problem means that it is most likely a Dijkstra problem.
 
-## Code
+To form the edges, we look at the possible movements that we can take. Firstly, we can jump off of the current rectangle and potentially reach a new rectangle. Secondly, we can move along our current rectangle.
 
+The vertices in this case would be every point on our rectangle, and the edges in this case would be every point on the exterior of the rectangle. However, for the full problem this would be too many (up to 1e9) points!
+
+We can use the technique of coordinate compression to reduce the amount of points. As we have $n \leq 1000$, there can be at most $2000$ horizontal and $2000$ vertical points, for a total of at most $4000000$ points. And similarly, there can be at most $4000000$ edges. This *barely* fits into an $O(m \lg n)$ time frame worst worst case. There can probably be more optimizations but this works.
```

Comment:
```suggestion
We can use the technique of coordinate compression to reduce the number of points. As we have $n \leq 1000$, there can be at most $2000$ horizontal and $2000$ vertical points, for a total of at most $4000000$ points. And similarly, there can be at most $4000000$ edges. This *barely* fits into an $O(m \log n)$ time frame worst case. There can probably be more optimizations but this works.
```

Maybe also mention how we only care about "critical" points? i.e. we don't care about lines starting and ending in the middle of the rectangles' sides

Reply:
lg is log base 2 btw